sirepo_detector.py

```diff
@@ -257,12 +249,18 @@ def select_optic(self, name):
     """
     def create_parameter(self, name):
         real_name = "sirepo_" + name
-        if self.field0 is None:
-            self.field0 = real_name
-        else:
-            self.field1 = real_name
+        ct = 0
+        while 'field' + str(ct) in self.fields.keys():
+            ct += 1
+        fieldkey = 'field' + str(ct)
+        parentkey = 'par' + str(ct)
```

Comment:
f-strings may be better in the 4 lines above. That way you will not need to wrap `ct` into `str(...)`